src/Symfony/Bridge/Doctrine/Form/ChoiceList/DoctrineChoiceLoader.php

```diff
@@ -146,7 +146,9 @@ public function loadChoicesForValues(array $values, $value = null)
 
         // Optimize performance in case we have an object loader and
         // a single-field identifier
-        if (null === $value && !$this->choiceList && $this->objectLoader && $this->idReader->isSingleId()) {
+        $optimize = null === $value || $isDefaultValue = is_array($value) && isset($value[0]) && $value[0] instanceof IdReader;
```

Comment:
no need to check for ``isset($value[0])`` actually. An array callable must always have it.

``$isDefaultValue`` seems unused here.

and if you want an even safer check, you may use ``$value[0] === $this->idReader`` (as passing a different idReader would also force us to disable the optimization as we could not ensure that ``isSingleId`` would be the right one below)